File: backend/app/schemas/meeting.py
"""
Pydantic schemas for Meeting API requests and responses.
"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class MeetingResponse(BaseModel):
    """Response schema for a single meeting."""
    id: str
    user_id: str
    title: str
    source_type: str
    status: str
    file_path: Optional[str] = None
    meeting_date: Optional[str] = None
    duration_seconds: Optional[int] = None
    created_at: Optional[str] = None

    class Config:
        from_attributes = True


class MeetingListResponse(BaseModel):
    """Response schema for listing meetings."""
    meetings: List[MeetingResponse]
    total: int

File: frontend/src/types/meeting.ts
export interface Meeting {
  id: string;
  user_id: string;
  title: string;
  source_type: "upload_text" | "upload_audio" | "zoom" | "gmeet";
  status: "pending" | "processing" | "completed" | "failed";
  file_path?: string;
  meeting_date?: string;
  duration_seconds?: number;
  created_at?: string;
}

export interface MeetingListResponse {
  meetings: Meeting[];
  total: number;
}
